engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java

```diff
@@ -392,15 +405,16 @@ protected Answer copyVolumeBetweenPools(DataObject srcData, DataObject destData)
                     return answer;
                 }
             } catch (Exception e) {
-                if (imageStore.exists(objOnImageStore)) {
+                if (objOnImageStore != null) {
                     objOnImageStore.processEvent(Event.OperationFailed);
                     imageStore.delete(objOnImageStore);
```

Comment:
if any volumes on store at this point, have to remove  them. check if you have to call _deleteVolumeOnSecondaryStore()_ in this case as well

Reply:
@Pearl1594 can you reply ^^

Reply:
It makes sense to delete the volume on secondary store on failure. Will add that - needs testing.

Reply:
@Pearl1594 any update on the changes / testing ?

Reply:
Haven't had the chance to get to testing this yet @sureshanaparti  - Will do it by end of this week.